### PR TITLE
Editor embeds merge in nitro embed

### LINKED ISSUE #256: [Editor] Embeds
As a user
I would like to see link previews and video embeds
In order to not have to click or leave the page to get the contents
---
We need to render `a` tags with a `data-url-embed` attribute an embed component. As a reference we have to look into the implementation of the todo-item node from the tiptap examples.
https://github.com/scrumpy/tiptap/blob/master/packages/tiptap-extensions/src/nodes/TodoItem.js
https://tiptap.scrumpy.io/todo-list

### PR DESCRIPTION
**Pull request by mattwr18**
### Issues

- fixes #256

### CODE REVIEW

**Comment by mattwr18** [comment]
does this look ok @roschaefer? I used `tomono`, but because I already had a repo with the name `Human-Connection`, I needed to remove it and recreate it with the merged commit history. This caused some small issues for me locally, but I think others won't face the same.

The files changed and commit history look good to me

**Comment by roschaefer** [comment]
@mattwr18 this looks good, but could sb. explain me why this is a separate service and not just another resolver in our backend?

@appinteractive maybe?

**Comment by mattwr18** [comment]
I think it could be, I thought the same thing myself @roschaefer

File: backend/src/schema/resolvers/embeds.spec.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+              title: 'Baby Loves Cat',
+              author: 'Merkley Family',
+              publisher: 'YouTube',
+              date: expect.any(String),
+              description:
+                'She’s incapable of controlling her limbs when her kitty is around. The obsession grows every day. Ps. That’s a sleep sack she’s in. Not a starfish outfit. Al...',
+              url: 'https://www.youtube.com/watch?v=qkdXAtO40Fo',
+              image: 'https://i.ytimg.com/vi/qkdXAtO40Fo/maxresdefault.jpg',
+              audio: null,
+              video: null,
```
Comment (roschaefer) [inline backend/src/schema/resolvers/embeds.spec.js]: `audio` and `video` was always `null` in all my tests...

File: backend/src/schema/resolvers/embeds.spec.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    author_url: 'https://www.youtube.com/channel/UC5P8yei950tif7UmdPpkJLQ',
+  }),
+)
+
+describe('Query', () => {
+  describe('embed', () => {
+    let embedAction
+
+    beforeEach(() => {
+      embedAction = async variables => {
```
Comment (mattwr18) [inline backend/src/schema/resolvers/embeds.spec.js]: :heart: the naming convention @roschaefer

File: backend/src/schema/resolvers/embeds.spec.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+    describe('given a Facebook link', () => {
+      beforeEach(() => {
+        fetch
+          .mockReturnValueOnce(Promise.resolve(new Response(HumanConnectionOrg)))
+          .mockReturnValueOnce(Promise.resolve('invalid json'))
+        variables = { url: 'https://www.facebook.com/HumanConnectionOrg/' }
+      })
+
+      it('does not crash if embed provider returns invalid JSON', async () => {
```
Comment (mattwr18) [inline backend/src/schema/resolvers/embeds.spec.js]: Any idea when providers return invalid JSONs and why?

File: backend/src/schema/resolvers/embeds.spec.js (hunk 1 above)
Comment (mattwr18) [reply backend/src/schema/resolvers/embeds.spec.js]: strange, that was what I observed as well

File: backend/src/schema/resolvers/embeds/scraper.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+// some providers allow a format parameter
+// we need JSON
+oEmbedProvidersFile = oEmbedProvidersFile.replace('{format}', 'json')
+
+const oEmbedProviders = JSON.parse(oEmbedProvidersFile)
+
+const fetchEmbed = async url => {
+  const provider = oEmbedProviders.find(provider => {
+    return provider.provider_url.includes(url.hostname)
```
Comment (mattwr18) [inline backend/src/schema/resolvers/embeds/scraper.js]: can you explain where you got this long list of providers? how does this work actually? we added one of @Tirokk personal sites and it returned the metadata, but not sure how any of the hard-coded providers include his url hostname :thinking:

File: backend/src/schema/resolvers/embeds/scraper.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  return {
+    sources: ['resource'],
+    ...resource,
+  }
+}
+
+export default async function scrape(url) {
+  url = new URL(url)
+  if (url.hostname === 'youtu.be') {
+    // replace youtu.be to get proper results
```
Comment (mattwr18) [inline backend/src/schema/resolvers/embeds/scraper.js]: thank you for your strategic use of comments @roschaefer

File: backend/src/schema/resolvers/users.js [hunk 6]
```diff
@@ -36,16 +37,6 @@ const count = obj => {
...
-const undefinedToNull = list => {
-  const resolvers = {}
-  list.forEach(key => {
-    resolvers[key] = async (parent, params, context, resolveInfo) => {
-      return typeof parent[key] === 'undefined' ? null : parent[key]
-    }
-  })
-  return resolvers
-}
-
```
Comment (mattwr18) [inline backend/src/schema/resolvers/users.js]: Nice refactoring :smile_cat:

File: backend/src/schema/resolvers/embeds.spec.js (hunk 1 above)
Comment (appinteractive) [reply backend/src/schema/resolvers/embeds.spec.js]: Is should be filmed when there is video or audio contend in an article as far as I understand metascraper.

File: backend/src/schema/resolvers/embeds/scraper.js (hunk 4 above)
Comment (appinteractive) [reply backend/src/schema/resolvers/embeds/scraper.js]: oEmbed is a specefication which Sites can use for embeds but don’t have to. It’s a way I enhanced the metascraper results with data directly from the providers. There is an official oEmbed list which contains a lot of sites which supports that. Does that answer your question?

File: backend/src/schema/resolvers/embeds.spec.js (hunk 1 above)
Comment (appinteractive) [reply backend/src/schema/resolvers/embeds.spec.js]: *filled

**Pull request merged**